backend/src/schema/resolvers/registration.js

```diff
@@ -0,0 +1,107 @@
+import { UserInputError } from 'apollo-server'
+import uuid from 'uuid/v4'
+import { neode } from '../../bootstrap/neo4j'
+import fileUpload from './fileUpload'
+import encryptPassword from '../../helpers/encryptPassword'
+
+const instance = neode()
+
+/*
+ * TODO: remove this function as soon type `User` has no `email` property
+ * anymore
+ */
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-03T17:58:30Z" title="Wednesday, July 3rd 2019, 7:58:30 pm +02:00">Jul 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,107 @@
+import { UserInputError } from 'apollo-server'
+import uuid from 'uuid/v4'
+import { neode } from '../../bootstrap/neo4j'
+import fileUpload from './fileUpload'
+import encryptPassword from '../../helpers/encryptPassword'
+
+const instance = neode()
+
+/*
+ * TODO: remove this function as soon type `User` has no `email` property
+ * anymore
+ */
```


what does this mean? user's will just have a username registered with our site? how would we send them an email for password reset, etc...

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-03T20:06:05Z" title="Wednesday, July 3rd 2019, 10:06:05 pm +02:00">Jul 3, 2019</time>_

---


I made a refactoring to the type `User` and left it at a dangerously in-between state. So, right now, the `User` has an attribute `email`. The idea of the the signup is, however, that you add an `EmailAddress` *without* a user account and then create the user account and connect it with the orphaned email address. This `email` attribute is a left-over. It should be completely removed at some point. But I think it's OK for now to keep it, it doesn't break anything and since we're free to create our own resolvers, we can also have a `User.email` resolver, which basically just `MATCH(u:User)-[:PRIMARY_EMAIL]->(e:EmailAddress)` pulls in the email. This is also a preparation for having multiple email addresses.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-03T21:28:54Z" title="Wednesday, July 3rd 2019, 11:28:54 pm +02:00">Jul 3, 2019</time>_

---


great, thanks for the explanation